src/ucp/rma/flush.c

```diff
@@ -487,61 +498,69 @@ static ucs_status_ptr_t ucp_worker_flush_nb_internal(ucp_worker_h worker,
         return UCS_STATUS_PTR(status);
     }
 
-    req = ucp_request_get(worker);
-    if (req == NULL) {
-        return UCS_STATUS_PTR(UCS_ERR_NO_MEMORY);
-    }
+    req = ucp_request_get_param(worker, param,
+                                {return UCS_STATUS_PTR(UCS_ERR_NO_MEMORY);});
 
-    req->flags                   = req_flags;
+    req->flags                   = 0;
     req->status                  = UCS_OK;
     req->flush_worker.worker     = worker;
-    req->flush_worker.cb         = cb;
     req->flush_worker.comp_count = 1; /* counting starts from 1, and decremented
                                          when finished going over all endpoints */
     req->flush_worker.prog_id    = UCS_CALLBACKQ_ID_NULL;
     req->flush_worker.next_ep    = ucs_list_head(&worker->all_eps,
                                                  ucp_ep_ext_gen_t, ep_list);
 
+    if (param->op_attr_mask & UCP_OP_ATTR_FIELD_CALLBACK) {
+        ucp_request_set_callback(req, flush_worker.cb, param->cb.send,
```

Comment:
maybe do a common macro instead of `ucp_request_set_send_callback_param`

Reply:
why?

Reply:
ucp_request_set_send_callback_param

Reply:
unhandled: use ucp_request_set_send_callback_param